clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/NonClusteredEagerEvictionScheduler.java

```diff
@@ -0,0 +1,193 @@
+
+    NonClusteredEagerEvictionScheduler(BeanFactory<I, T> factory, BeanGroupEvictor<I> evictor, ScheduledExecutorService executor, Duration idleTimeout, CommandDispatcherFactory dispatcherFactory, String dispatcherName) {
+        this.factory = factory;
+        this.executor = executor;
+        this.idleTimeout = idleTimeout;
+        this.expirationTracker = new ExpirationTracker<>(idleTimeout);
+        this.dispatcher = dispatcherFactory.createCommandDispatcher(dispatcherName + "/eviction", evictor);
+    }
+
+    @Override
+    public void accept(I id) {
+        this.expirationTracker.forget(id);
+    }
+
+    @Override
+    public void schedule(I id) {
+        BeanEntry<I> entry = this.factory.findValue(id);
+        if (entry != null) {
+            InfinispanEjbLogger.ROOT_LOGGER.tracef("Scheduling stateful session bean %s to passivate in %s", id, this.idleTimeout);
+            expirationTracker.trackExpiration(id);
+            if (!groupIdMap.containsKey(id)) {
+                groupIdMap.put(id, entry.getGroupId());
+            }
+            // Make sure the map insertion happens before map removal (during task execution).
+            if (evictionTask == null)
+            synchronized (this) {
+                if (evictionTask == null) {
+                    Runnable task = new EvictionTask();
+                    evictionTask = this.executor.schedule(task, this.idleTimeout.toMillis(), TimeUnit.MILLISECONDS);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void schedule(I id, ImmutableBeanEntry<I> entry) {
+        schedule(id);
+    }
+
+    @Override
+    public void prepareRescheduling(I id) {
+        expirationTracker.invalidateExpiration(id);
+    }
+
+    @Override
+    public void cancel(I id) {
+        expirationTracker.forget(id);
+        groupIdMap.remove(id);
+    }
+
+    @Override
+    public void cancel(Locality locality) {
+        for (I id: this.expirationTracker.getTrackedIds()) {
+            if (Thread.currentThread().isInterrupted()) break;
+            if (!locality.isLocal(id)) {
+                this.cancel(id);
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        final Future<?> task;
+        synchronized (this) {
+            if (this.evictionTask == null) {
+                return;
+            }
+            task = evictionTask;
+
+        }
+        task.cancel(false);
+        if (!task.isDone()) {
+            try {
+                task.get();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException e) {
+                // Ignore
+            }
+        }
+        this.dispatcher.close();
+    }
+
+    @Override
+    public void evict(I id) {
+        try {
+            // Cache eviction is a local operation, so we need to broadcast this to the cluster
+            this.dispatcher.executeOnGroup(new EvictCommand<>(id));
```

Comment:
Given that this is now a local-only scheduler, there is no need to dispatch an eviction command.